etc/network/interfaces.d/30_non-qubes-whonix

```diff
@@ -38,6 +38,7 @@ iface lo inet loopback
 ## https://phabricator.whonix.org/T559
 #auto eth0
 #iface eth0 inet dhcp
+iface eth0 inet6 auto
```

Comment:
In principle yes, but this doesn't work with our network setup. As of now VirtualBox doesn't support IPv6 NAT. libvirt seems to support it, by adding a static ipv6 address to the network config + manually adding Masquerading rules. libvirt automatically adds this only for ipv4, but not for ipv6, so if we want to make use of this, we'd have to add an additional hook to configure the firewall accordingly. The auto config I used above was working when I switched from NAT to using a bridged interface. This directly exposed the gateway on the network though, which we probably don't want to do.

Reply:
If I'm understanding you correctly, that means that with all of the IPv6 code, this will *not* work under VirtualBox, and will require tweaks to the libvirt configuration for it to work on libvirt, right? (I'm sure you explained the issue quite well, but I'm not a networking expert so I'm unsure what's supposed to or not supposed to work. I did test this and all the other PRs together on a couple pairs of Whonix VMs, one running under VirtualBox and the other running under libvirt. In both instances, using cURL with the `--ipv6` switch returned `curl: (7) Failed to connect to IPADDR port 9050 after 0 ms: Couldn't connect to server`.)

Reply:
FWIW I tried adding an IPv6 network in libvirt and couldn't get this to work. In particular, I set it up so that I have a `Whonix-ExternalIPv6` network attached only to Whonix-Gateway, configured like this:

```xml
<network>
  <name>Whonix-ExternalIPv6</name>
  <uuid>a88cf385-b387-419a-9acb-46f1ca18d3ad</uuid>
  <forward mode="nat">
    <nat>
      <port start="1024" end="65535"/>
    </nat>
  </forward>
  <bridge name="virbr3" stp="on" delay="0"/>
  <mac address="52:54:00:50:0b:49"/>
  <dns enable="no"/>
  <ip address="10.0.2.2" netmask="255.255.255.0">
  </ip>
  <ip family="ipv6" address="fd19:c33d:88bc::1" prefix="96">
  </ip>
</network>
```

(That was the best I could figure looking at the existing libvirt Whonix network config and reading through the source code.) This doesn't appear to work - `curl --ipv6 --proxy socks5h://[::1]:9050 https://v6.ipv6test.app/` running on the workstation VM immediately fails because it can't connect to the proxy. I furthermore tried disabling the whonix firewall on both VMs and then tried using netcat with IPv6 to transfer strings back and forth. Ultimately no data was able to move between the two VMs. I assume I'm probably just doing something wrong here, but I can't quite figure this out.